apps/condo/domains/user/schema/ConfirmPhoneAction.js

```diff
@@ -0,0 +1,245 @@
+/**
+ * Generated by `createschema user.ConfirmPhoneAction 'phone:Text;token:Text;smsCode:Integer;smsCodeRequestedAt:DateTimeUtc;smsCodeExpiresAt:DateTimeUtc;retries:Integer;isPhoneVerified:Checkbox;requestedAt:DateTimeUtc;expiresAt:DateTimeUtc;completedAt:DateTimeUtc;'`
+ */
+const { v4: uuid } = require('uuid') 
+const { Text, Integer, Checkbox, DateTimeUtc } = require('@keystonejs/fields')
+const { GQLListSchema, GQLCustomSchema } = require('@core/keystone/schema')
+const { historical, uuided, softDeleted } = require('@core/keystone/plugins')
+const { SENDER_FIELD, DV_FIELD } = require('@condo/domains/common/schema/fields')
+const access = require('@condo/domains/user/access/ConfirmPhoneAction')
+const { normalizePhone } = require('@condo/domains/common/utils/phone')
+const isEmpty = require('lodash/isEmpty')
+const { sendMessage } = require('@condo/domains/notification/utils/serverSchema')
+const { ConfirmPhoneAction: ConfirmPhoneActionGQL, generateSmsCode } = require('@condo/domains/user/utils/serverSchema')
+
+const { 
+    CONFIRM_PHONE_ACTION_EXPIRED,
+    CONFIRM_PHONE_SMS_CODE_EXPIRED,
+    CONFIRM_PHONE_SMS_CODE_VERIFICATION_FAILED, 
+    CONFIRM_PHONE_SMS_CODE_MAX_RETRIES_REACHED, 
+} = require('@condo/domains/user/constants/errors')
+
+const { COUNTRIES, RUSSIA_COUNTRY } = require('@condo/domains/common/constants/countries')
+const { SMS_VERIFY_CODE_MESSAGE_TYPE } = require('@condo/domains/notification/constants')
+
+const { 
+    SMS_CODE_LENGTH, 
+    SMS_CODE_TTL, 
+    CONFIRM_PHONE_ACTION_EXPIRY, 
+    CONFIRM_PHONE_SMS_MAX_RETRIES,
+} = require('@condo/domains/user/constants/common')
+
+const ConfirmPhoneAction = new GQLListSchema('ConfirmPhoneAction', {
+    schemaDoc: 'User confirm phone actions is used before registration starts',
+    fields: {
+        dv: DV_FIELD,
+        sender: SENDER_FIELD,
+        phone: {
+            schemaDoc: 'Phone. In international E.164 format without spaces',
+            type: Text,
+            kmigratorOptions: { null: true, unique: false },
+            hooks: {
+                resolveInput: ({ resolvedData }) => {
+                    return normalizePhone(resolvedData['phone'])
+                },
+            },
+        },
+        token: {
+            schemaDoc: 'Unique token to complete confirmation',
+            type: Text,
+            isUnique: true,
+            isRequired: true,
+        },
+        smsCode: {
+            schemaDoc: 'Last sms code sent to user',
+            type: Integer,
+            length: SMS_CODE_LENGTH,
+        },
+        smsCodeRequestedAt: {
+            schemaDoc: 'Time when sms code was requested',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        smsCodeExpiresAt: {
+            schemaDoc: 'Time when sms code becomes not valid',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        retries: {
```

Comment:
Should we also store information about number of trials to resend SMS?

Reply:
There will be one more pr on security subject with recaptcha3 and server-side timeout locks